lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -236,8 +238,21 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			log.Infof("Sending die event for container %s - code: %s", vc.ContainerID, code)
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
-
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+
+			// auto-remove if required
+			if vc.HostConfig.AutoRemove {
+				config := &types.ContainerRmConfig{
+					ForceRemove:  true,
+					RemoveVolume: true,
```

Comment:
I hope, but do not know, that we correctly leave volumes alone if they're attached to other containers.
The only time this is going to be relevant for us currently is with NFS volumes.
@matthewavery @fdawg4l could one of you comment?